### DISCUSSION TITLE
nodata for writter function in auxil.py

### DISCUSSION THREAD

**Discussion by gr4n0t4** [johntruckenbrodt/pyroSAR#230] [Ideas]
Hi, the "noData" value for the write function is set to '0' 
For ERS1/2 & ASAR sensor it would work best with 'NaN' instead of '0', I'm being told that could be an issue for the "log scaled images" in S1
So to implement this option would it be best to add a optional parameter to the writer function or to check the sensor inside as it is done for writing the SAFE file?
Thank you

**Comment by johntruckenbrodt** [comment]
Hi @gr4n0t4. True, this value is hard-coded for [pyroSAR.snap.auxil.writer](https://pyrosar.readthedocs.io/en/latest/api/snap.html#pyroSAR.snap.auxil.writer). To optionally change the output nodata value would make sense. However, the internally used [gdal.TranslateOptions](https://gdal.org/api/python/osgeo.gdal.html#osgeo.gdal.TranslateOptions) just has a `noData` parameter which applies to both source and target. Thus, modifying it would omit the source nodata value of 0, which is SNAP's standard. The thing is, SNAP does not write the nodata value in the ENVI HDR files (which are used by both SNAP's BEAM-DIMAP and ENVI output format) so it cannot be read in GDAL when modifying `noData` (perhaps GDAL assumes 0 if ot cannot read a value though, not quite sure). Easy modifcation of the HDR file to add the nodata value is possible with [spatialist.envi](https://spatialist.readthedocs.io/en/latest/spatialist.html#module-spatialist.envi):
```python
from spatialist.envi import HDRobject
with HDRobject(hdrfile) as hdr:
    hdr.data_ignore_value = 0
    hdr.write(hdrfile)
```
Once this is done, passing `NaN` to `gdal.Translate` should work fine.

FYI, I have recently restructured the [S1_NRB](https://github.com/SAR-ARD/S1_NRB) processor to no longer make use of [pyroSAR.snap.util.geocode](https://pyrosar.readthedocs.io/en/latest/api/snap.html#pyroSAR.snap.util.geocode). See [S1_NRB.snap](https://s1-nrb.readthedocs.io/en/latest/api.html#module-S1_NRB.snap).

**Reply by gr4n0t4** [reply]
Thanks for your reply. We will investigate your solutions
